### PR TITLE
Features of : Add Note with Color, Update Existing Note, Pin Notes 

### PR DESCRIPTION
**Pull request by dev-enforced**
In this PR one can find the next preview of the Notes App which consists of filling up the following details of the Note:
->Title
-> Content to be passed inside the embedded text editor
-> Color of the upcoming Note
-> Priority value of the note to be accessed through the dropdown

After creating a new note the user can either update the note based on the above details through a pencil shaped icon provided on the NOTE CARD !!
The user can even pin the notes to make sure that they remembers the note!!
Here is the live preview link for the same: https://notes-app-recreated-git-crud-operations-not-1c0503-dev-enforced.vercel.app/

### CODE REVIEW

File: src/components/NoteInput/NoteInput.jsx [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
-            <input type="text" className='note-input p-3' placeholder='ENTER TITLE' />
+        <form className='note-input-container' onSubmit={(e) => {
+            e.preventDefault()
+            const checkProductExistence = notesList.some((everyProduct) => everyProduct._id === noteData._id);
+            if (checkProductExistence) {
+                updateExistingNote(noteData);
+            } else {
+                addNewNote(noteData);
+            }
+            dispatchNoteData({ type: "RESET_INPUT_DATA" })
```
Comment (mariyasada) [inline src/components/NoteInput/NoteInput.jsx]: you can create separate function for this

File: src/components/NoteItem/NoteItem.css [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.card-actions .btn{
+/* .card-actions .btn{
```
Comment (mariyasada) [inline src/components/NoteItem/NoteItem.css]: remove this commented code if not necessary

File: src/components/PriorityDropdown/PriorityDropdown.jsx [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+                    dispatchNoteData({ type: "UPDATE_PRIORITY", payload: "High" });
+                    togglePriorityDropdownStyle();
+                }} className='py-2 px-4'>High</li>
+                <li onClick={() => {
+                    dispatchNoteData({ type: "UPDATE_PRIORITY", payload: "Medium" });
+                    togglePriorityDropdownStyle();
+                }} className='py-2 px-4'>Medium</li>
+                <li onClick={() => {
+                    dispatchNoteData({ type: "UPDATE_PRIORITY", payload: "Low" });
+                    togglePriorityDropdownStyle();
```
Comment (mariyasada) [inline src/components/PriorityDropdown/PriorityDropdown.jsx:27]: liked that you have created reducer for this

File: src/services/Notes/updateExistingNoteService.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+const updateExistingNoteService = async (givenNote, tokenValue) => {
```
Comment (bhavyaj12) [inline src/services/Notes/updateExistingNoteService.js:2]: For updating a better user experience would be to display the update form as a modal. Updating in the same form can be a little confusing

**Review by mariyasada — APPROVED** [review]
everything looks good to me. the pop-up box of priority looks cool

**Review by bhavyaj12 — APPROVED** [review]
Seems to work great. Have added a comment. Logic is good. Good use of Rich Text Editor. 
You can add an edge case of note not being added if content is empty, because user might accidentally just add a title. This is already handled for title in your form, you can handle for note content too.

File: src/components/NoteInput/NoteInput.jsx (hunk 1 above)
Comment (dev-enforced) [reply src/components/NoteInput/NoteInput.jsx]: Extracting a function for the submission handler can involve passing multiple parameters such as the dispatch and event ones into it so keeping it in the form component is more useful to me than creating a new function for generating the values!!

File: src/components/NoteItem/NoteItem.css (hunk 2 above)
Comment (dev-enforced) [reply src/components/NoteItem/NoteItem.css]: done

File: src/services/Notes/updateExistingNoteService.js (hunk 4 above)
Comment (dev-enforced) [reply src/services/Notes/updateExistingNoteService.js:2]: Didn't knew this perspective of update Form with a separate modal. Will try this out while refactoring the app at the end. Thank you @bhavyaj12 for this suggestion!!

File: src/components/NoteInput/NoteInput.jsx (hunk 1 above)
Comment (dev-enforced) [reply src/components/NoteInput/NoteInput.jsx]: done the changes

**Pull request merged**